app/src/main/java/chat/rocket/android/chatroom/adapter/ActionsAttachmentViewHolder.kt

```diff
@@ -0,0 +1,105 @@
+package chat.rocket.android.chatroom.adapter
+
+import android.view.LayoutInflater
+import android.view.View
+import android.view.ViewGroup
+import androidx.recyclerview.widget.RecyclerView
+import chat.rocket.android.chatroom.uimodel.ActionsAttachmentUiModel
+import chat.rocket.android.emoji.EmojiReactionListener
+import chat.rocket.core.model.attachment.actions.Action
+import chat.rocket.core.model.attachment.actions.ButtonAction
+import kotlinx.android.synthetic.main.item_actions_attachment.view.*
+import androidx.recyclerview.widget.LinearLayoutManager
+import chat.rocket.android.R
+import chat.rocket.android.util.TimberLogger
+import com.facebook.drawee.backends.pipeline.Fresco
+import com.facebook.drawee.view.SimpleDraweeView
+import com.google.android.material.button.MaterialButton
+
+class ActionsAttachmentViewHolder(
+        itemView: View,
+        listener: ActionsListener,
+        reactionListener: EmojiReactionListener? = null,
+        var actionAttachmentOnClickListener: ActionAttachmentOnClickListener
+) : BaseViewHolder<ActionsAttachmentUiModel>(itemView, listener, reactionListener) {
+
+    init {
+        with(itemView) {
+            setupActionMenu(actions_attachment_container)
+        }
+    }
+
+    override fun bindViews(data: ActionsAttachmentUiModel) {
+        val actions = data.actions
+        TimberLogger.debug("no of actions : ${actions.size} : $actions")
+        with(itemView) {
+            title.text = data.title ?: ""
+            actions_list.layoutManager = LinearLayoutManager(itemView.context)
+            actions_list.adapter = ActionsListAdapter(actions, actionAttachmentOnClickListener)
+        }
+    }
+}
+
+interface ActionAttachmentOnClickListener {
+    fun onActionClicked(action: Action)
+}
+
+class ActionsListAdapter(actions: List<Action>, var actionAttachmentOnClickListener: ActionAttachmentOnClickListener) : RecyclerView.Adapter<ActionsListAdapter.ViewHolder>() {
```

Comment:
I think I'd prefer using a new package and multiple files instead of multiple classes and inner classes on the same file